tests/jballerina-unit-test/src/test/java/org/ballerinalang/test/record/MapToRecordAssignabilityTest.java

```diff
@@ -97,6 +97,8 @@ public void testComplexSubtyping2() {
                 {"testBasicDataTypeMaps"},
                 {"testInclusiveRecordTypes"},
                 {"testRecordsWithOptionalFields"},
+                {"testImmutableJsonMappingToExclusiveRecord"},
+                {"testImmutableJsonMappingToInclusiveRecord"},
         };
```

Comment:
Since we are testing casting here, IMO, we should move this to a type cast test.

Reply:
done